sw/device/tests/uart_tx_rx_test.c

```diff
@@ -120,6 +120,8 @@ static volatile uart_cfg_params_t uart_cfg;
  */
 static volatile bool exp_uart_irq_tx_watermark;
 static volatile bool uart_irq_tx_watermark_fired;
+static volatile bool exp_uart_irq_tx_empty;
```

Comment:
This looks perfectly reasonable, but I wonder whether it makes sense to split the additions to the test into a separate commit? I don't think there's any reason they have to be atomic with the RTL change.